spring-pulsar-spring-cloud-stream-binder/src/test/java/org/springframework/pulsar/spring/cloud/stream/binder/PulsarTopicProvisionerTests.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.spring.cloud.stream.binder;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+import org.springframework.cloud.stream.binder.ExtendedConsumerProperties;
+import org.springframework.cloud.stream.binder.ExtendedProducerProperties;
+import org.springframework.cloud.stream.provisioning.ConsumerDestination;
+import org.springframework.cloud.stream.provisioning.ProducerDestination;
+import org.springframework.pulsar.core.PulsarAdministration;
+import org.springframework.pulsar.core.PulsarTopic;
+import org.springframework.pulsar.spring.cloud.stream.binder.properties.PulsarBinderConfigurationProperties;
+import org.springframework.pulsar.spring.cloud.stream.binder.properties.PulsarConsumerProperties;
+import org.springframework.pulsar.spring.cloud.stream.binder.properties.PulsarProducerProperties;
+import org.springframework.pulsar.spring.cloud.stream.binder.provisioning.PulsarTopicProvisioner;
+
+/**
+ * @author Soby Chacko
+ */
+public class PulsarTopicProvisionerTests {
+
+	@Test
+	void provisionThroughProducerBindingWithDefaultPartitioning() {
+		PulsarBinderConfigurationProperties pulsarBinderConfigurationProperties = new PulsarBinderConfigurationProperties();
+		PulsarAdministration pulsarAdministration = mock(PulsarAdministration.class);
+		PulsarTopicProvisioner pulsarTopicProvisioner = new PulsarTopicProvisioner(pulsarAdministration,
+				pulsarBinderConfigurationProperties);
+		ExtendedProducerProperties<PulsarProducerProperties> properties = new ExtendedProducerProperties<>(
+				new PulsarProducerProperties());
+		ProducerDestination producerDestination = pulsarTopicProvisioner.provisionProducerDestination("foo",
+				properties);
+		verifyAndAssert(pulsarAdministration, producerDestination.getName(), "foo", 1);
+	}
+
+	private static void verifyAndAssert(PulsarAdministration pulsarAdministration, String producerDestination,
+			String foo, int expected) {
```

Comment:
I think renaming the args in the `actual / expected` pattern makes this easier to follow:
```java
producerDestination -> actualProducerDestination
foo -> expectedProducerDestination
expect -> expectedPartitionCount
```
wdyt?

Reply:
Good idea.